### ISSUE TITLE
Will SalatMongoCursor leak memory if they aren't closed?

### ISSUE THREAD

**Issue by zon** [salat/salat#121]
The wiki's DAO example transforms a SalatMongoCursor into a list and does not close it afterwards.

```scala
val results = AlphaDAO.find(ref = MongoDBObject("_id" -> MongoDBObject("$gte" -> 2)))
        .sort(orderBy = MongoDBObject("_id" -> -1)) // sort by _id desc
        .skip(1)
        .limit(1)
        .toList // yay!
```

Is this good practice or will the program end up leaking memory?

**Comment by noahlz** [comment]
(Triaging old issues...)

Is this related to Java Mongo Driver behavior where it keeps cursors open and closes them upon Finalization?
https://jira.mongodb.org/browse/JAVA-610

Novus Partners ran Salat for years with code in this manner, and memory leaks due to floating cursors were never a problem, as far as I can recall.

**Comment by noahlz** [comment]
I also found https://jira.mongodb.org/browse/SERVER-3295?focusedCommentId=38271&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-38271

> cursors get closed when:
> 
>    10 minutes of inactivity
>    client consume all data
>    client closes once a user is done with it

Closing this issue, but future readers are welcome to add more comments / details, and/or open a new issue!
